src/index.ts

```diff
@@ -1,103 +1,127 @@
-import * as visit from 'unist-util-visit';
-import {Code, Parent} from 'mdast';
-import {Node} from 'unist';
+import visit, {Visitor} from 'unist-util-visit';
+import {Code} from 'mdast';
+import {Parent} from 'unist';
 import {Options as PrettierOptions, format} from 'prettier';
+import {
+  Transformations,
+  builtinTransformations,
+  createTransformationApplicator
+} from './custom-transformations';
 import {Transformer} from 'unified';
 import {transformSync} from '@babel/core';
 
-function insertAt(
-  array: Parent['children'],
-  index: number,
-  item: Code
-): Parent['children'] {
-  return [...array.slice(0, index), item, ...array.slice(index)];
-}
-
-function isReactComponent(node: Node, tag: string): boolean {
-  return node && node.type === 'jsx' && node.value === tag;
-}
-
-const PRESERVE_COMMENT = '// preserve-line';
-const PRESERVE_PATTERN = new RegExp(`\\s${PRESERVE_COMMENT}$`);
-
-function removePreserveComment(line: string): string {
-  return line.replace(PRESERVE_PATTERN, '');
-}
-
 interface Options {
-  wrapperComponent?: string;
+  /**
+   * Set if you want to throw on error, otherwise it logs to stderr
+   */
   throwOnError?: boolean;
+  /**
+   * Custom prettier options for the formatting step
+   */
   prettierOptions?: PrettierOptions;
+  /**
+   * Transform the code during differnet stages
+   */
+  customTransformations?: Transformations;
+  /**
+   * Determine if a particular ts/tsx code block should be skipped.
+   * @example In MDX, you could check for a particular wrapping component
+   */
+  shouldSkip?: (
+    node: Code,
+    parent: Parent<any>,
+    index: number,
+    meta?: string
+  ) => boolean;
 }
 
-export = function remarkTypescript({
-  wrapperComponent,
+/**
+ * Transforms ts and tsx code blocks into their JS counterparts for multi-code block automation
+ * @param param options
+ * @returns Transformer
+ */
+export default function remarkTypescript({
   throwOnError,
-  prettierOptions = {}
+  prettierOptions = {},
+  customTransformations = [],
+  shouldSkip
```

Comment:
Could we simplify the API for `shouldSkip` to accept either a boolean or a function that returns a boolean? And if just a boolean is passed in, we use a default `shouldSkip` function (similar to the one in the test file). But if the user wants to use their own custom skip function, they can still pass that in instead.

Reply:
I'm not sure what the use case for a `shouldSkip` boolean would be. If you're passing `shouldSkip: true`, then why use this plugin at all?

Reply:
My understanding is that it will transpile every TS code block and this is an alternate approach to the current method of using 'wrapperComponent' to signify which blocks to transpile vs skip over? Cc @BlenderDude

Reply:
Mm I misunderstood. I was thinking that the function for 'shouldSkip' would have to get passed to the actual code blocks themselves instead of in the plug-in config 🙃